CMakeLists.txt

```diff
@@ -135,66 +135,74 @@ if(ASPECT_WITH_WORLD_BUILDER)
     MESSAGE(STATUS "World Builder not found. Using internal version.")
     SET(WORLD_BUILDER_SOURCE_DIR "${CMAKE_SOURCE_DIR}/contrib/world_builder/" CACHE PATH "" FORCE)
   ENDIF()
+  
+  # add source and include dirs:
+  MESSAGE(STATUS "Using World Builder version ${WORLD_BUILDER_VERSION} found at ${WORLD_BUILDER_SOURCE_DIR}.")
 
-  # Always include the header files.
   INCLUDE_DIRECTORIES("${WORLD_BUILDER_SOURCE_DIR}/include/")
-
   INCLUDE("${WORLD_BUILDER_SOURCE_DIR}/cmake/version.cmake")
-  MESSAGE(STATUS "Using World Builder version ${WORLD_BUILDER_VERSION} found at ${WORLD_BUILDER_SOURCE_DIR}.")
-
-  # add source and include dirs:
-  IF(WORLD_BUILDER_VERSION VERSION_LESS 0.5.0)
-    FILE(GLOB_RECURSE wb_files "${WORLD_BUILDER_SOURCE_DIR}/source/*.cc")
-  ELSE()
-    FILE(GLOB_RECURSE wb_files "${WORLD_BUILDER_SOURCE_DIR}/source/world_builder/*.cc")
-  ENDIF()
-  LIST(APPEND TARGET_SRC ${wb_files})
-  ADD_DEFINITIONS(-DWB_WITH_MPI)
-
-  # generate config.cc and include it:
-  IF(WORLD_BUILDER_VERSION VERSION_LESS 0.5.0)
-    CONFIGURE_FILE("${WORLD_BUILDER_SOURCE_DIR}/source/config.cc.in" "${CMAKE_BINARY_DIR}/source/world_builder/config.cc" @ONLY)
-    LIST(INSERT TARGET_SRC 0 "${CMAKE_BINARY_DIR}/source/world_builder/config.cc")
-    FILE(REMOVE "${CMAKE_BINARY_DIR}/include/world_builder/config.h")
-  ELSE()
-    CONFIGURE_FILE("${WORLD_BUILDER_SOURCE_DIR}/include/world_builder/config.h.in" "${CMAKE_BINARY_DIR}/include/world_builder/config.h" @ONLY)
-    INCLUDE_DIRECTORIES("${CMAKE_BINARY_DIR}/include/")
-    FILE(REMOVE "${CMAKE_BINARY_DIR}/source/world_builder/config.cc")
-  ENDIF()
 
-  # Move some file to the end for unity builds to make sure other file come
-  # "before". Note: The current design keeps all ASPECT files (including
-  # ASPECT_UNITY_LAST files) before all GWB files. Mixing them will causes
-  # many issues with non-unique namespace names like Utilities.
-  IF(WORLD_BUILDER_VERSION VERSION_LESS 0.5.0)
-    SET(UNITY_WB_LAST_FILES
-    "${WORLD_BUILDER_SOURCE_DIR}/source/parameters.cc")
+  IF(WORLD_BUILDER_VERSION VERSION_GREATER_EQUAL 0.5.0 AND NOT WORLD_BUILDER_VERSION_LABEL MATCHES "pre")
+    # Disable WorldBuilder messages less important than ERROR
+    SET (CMAKE_MESSAGE_LOG_LEVEL ERROR)
+    ADD_SUBDIRECTORY("${WORLD_BUILDER_SOURCE_DIR}" ${CMAKE_BINARY_DIR}/world_builder/ SYSTEM)
+        SET (CMAKE_MESSAGE_LOG_LEVEL STATUS)
+    INCLUDE_DIRECTORIES("${CMAKE_BINARY_DIR}/world_builder/include/")
   ELSE()
-    SET(UNITY_WB_LAST_FILES
-    "${WORLD_BUILDER_SOURCE_DIR}/source/world_builder/parameters.cc")
-  ENDIF()
+    IF(WORLD_BUILDER_VERSION VERSION_LESS 0.5.0)
```

Comment:
If we bundle 0.5, we might as well set the minimum supported version to 0.5.

Reply:
Because there may be a lot of users out there with custom world builder files and versions below 0.5 release (us for example). World builder files for version 0.4 do not work with 0.5, and requiring 0.5 forces a lot of people to upgrade their workflow immediately after release of world builder 0.5. I would rather wait until the next ASPECT release before we do that.

Reply:
Are you sure it's worth it for us to be forward and backward compatible? If a user updates to a new aspect release we can also expect that they are willing to upgrade WB. There is value in bundling the oldest supported version to improve test coverage.

Anyways, not a big deal if you feel confident this will still work..

Reply:
Well we bundled 0.4.0 up to now, so we are sure it works at  the moment. I am ok dropping support for World Builder 0.4.0 right after ASPECT's release, but I would like to keep it for the release. Mostly, to not surprise users who are currently using ASPECT's development  environment  and WB 0.4.0, then update the ASPECT version by a few days and suddenly need to use the bundled WB 0.5.0 immediately at the time of the WB release.